### PR TITLE
Adapt and document sweeper.py for gssproxy

### PR DESCRIPTION
**Pull request by frozencemetery**
Resolves: #207
Signed-off-by: Robbie Harwood <rharwood@redhat.com>

### CODE REVIEW

File: contrib/sweeper.py [hunk 1]
```diff
@@ -1,17 +1,24 @@
...
 # become littered with ccache files unless the accessed application cleans
 # them up itself.  This script will minimize ccache file proliferation by
 # removing any ccaches that have expired from the filesystem, and serves as an
 # example of how this cleaning can be performed.
 
+# gssproxy note: in order to sweep credentials, the sweeper needs to connect
+# to gssproxy as if it were mod_auth_gssapi.  In the configuration provided
+# with mod_auth_gssapi (80-httpd.conf), this just consists of matching the
+# gssproxy uid - so run it as the appropriate user (i.e., apache).  Custom
+# configurations require more configuration; this script is just an example.
```
Comment (simo5) [inline contrib/sweeper.py]: more configuration -> careful consideration about how the program connecting to gssproxy is matched to the proper configuration section

File: contrib/sweeper.py [hunk 2]
```diff
@@ -48,16 +55,21 @@ def should_delete(fname, t):
...
-        exit(1)
+    parser = argparse.ArgumentParser(description="Sweep expired ccaches")
+    parser.add_argument("-g", dest="gssproxy", action="store_true",
+                        help="is gssproxy in use (default: no)")
+    parser.add_argument("dirs", nargs='+')
+    args = parser.parse_args()
+
+    if args.gssproxy:
+        os.environ["GSS_USE_PROXY"] = "yes"
+        os.environ["GSSPROXY_BEHAVIOR"] = "REMOTE_FIRST"
```
Comment (simo5) [inline contrib/sweeper.py]: Should we use REMOTE_ONLY instead ?

**Comment by simo5** [comment]
```E: Unable to locate package python3-requests-gssapi```

lovely ...

File: contrib/sweeper.py (hunk 2 above)
Comment (frozencemetery) [reply contrib/sweeper.py]: Per gssproxy-mech(8), REMOTE_ONLY "is currently not fully implemented and therefor not supported."

**Comment by frozencemetery** [comment]
Yeah, that's just noise.  Added a commit to hide it.  (The actual error was flake8 related.)

**Pull request merged**